lib/ansible/modules/network/nxos/nxos_interface_ospf.py

```diff
@@ -67,9 +67,12 @@
         integer or the keyword 'default'.
   passive_interface:
     description:
-      - Setting to true will prevent this interface from receiving
-        HELLO packets.
-    type: bool
+      - Enable or disable passive-interface state on this interface.
+        true - (enable) Prevent OSPF from establishing an adjacency or
+                       sending routing updates on this interface.
+        false - (disable) Override global 'passive-interface default' for this interface.
+        default - Remove existing (enabled or disabled) passive-interface setting.
```

Comment:
@chrisvanheuveln Just If I understand correctly, passive-interface can be removed by setting it to default?
True/False is only for enabling and disabling right?

Reply:
@trishnaguha That's correct. In this case the *only* way to remove these configurations is to use the `default ip ospf passive-interface` syntax.

Reply:
@chrisvanheuveln Instead of adding `default` as a choice to `passive_interface`, IMO more efficient way to handle this would be to push the `default ip ospf passive-interface` command for `state` being `absent` as the intention is to remove the existing config.

Reply:
@trishnaguha My original concern was that that approach only applies when the intent is to remove *all* of the ospf interface configs. If the intent is to keep other ospf settings (e.g. authentication or network type) while only removing the passive config then it doesn't work.

But I'll concede that it's getting into corner case territory at this point. I ran it past Mike and we agreed to go with your suggestion instead to keep it simple. I'll rework the diffs. Thanks.